extensions/blocks/contact-form/components/jetpack-field-multiple.js

```diff
@@ -2,121 +2,173 @@
+		const newOptions = options.slice( 0 );
 
-	onChangeOption( key = null, option = null ) {
-		const newOptions = this.props.options.slice( 0 );
 		if ( null === option ) {
 			// Remove a key
 			newOptions.splice( key, 1 );
 			if ( key > 0 ) {
-				this.setState( { inFocus: key - 1 } );
+				setInFocus( key - 1 );
 			}
 		} else {
 			// update a key
 			newOptions.splice( key, 1, option );
-			this.setState( { inFocus: key } ); // set the focus.
+			setInFocus( key ); // set the focus.
 		}
-		this.props.setAttributes( { options: newOptions } );
-	}
+		setAttributes( { options: newOptions } );
+	};
+
+	const addNewOption = ( key = null ) => {
+		const newOptions = options.slice( 0 );
+		let newInFocus = 0;
 
-	addNewOption( key = null ) {
-		const newOptions = this.props.options.slice( 0 );
-		let inFocus = 0;
 		if ( 'object' === typeof key ) {
 			newOptions.push( '' );
-			inFocus = newOptions.length - 1;
+			newInFocus = newOptions.length - 1;
 		} else {
 			newOptions.splice( key + 1, 0, '' );
-			inFocus = key + 1;
+			newInFocus = key + 1;
 		}
 
-		this.setState( { inFocus: inFocus } );
-		this.props.setAttributes( { options: newOptions } );
-	}
-
-	render() {
-		const { type, instanceId, required, label, setAttributes, isSelected, id } = this.props;
-		let { options } = this.props;
-		let { inFocus } = this.state;
-		if ( ! options.length ) {
-			options = [ '' ];
-			inFocus = 0;
-		}
+		setInFocus( newInFocus );
+		setAttributes( { options: newOptions } );
+	};
 
-		return (
-			<Fragment>
-				<BaseControl
+	return (
+		<Fragment>
+			<BaseControl
+				id={ `jetpack-field-multiple-${ instanceId }` }
+				className="jetpack-field jetpack-field-multiple"
+				label={
+					<JetpackFieldLabel
+						required={ required }
+						label={ label }
+						setAttributes={ setAttributes }
+						isSelected={ isSelected }
+						resetFocus={ () => setInFocus( null ) }
+					/>
+				}
+			>
+				<ol
+					className="jetpack-field-multiple__list"
 					id={ `jetpack-field-multiple-${ instanceId }` }
-					className="jetpack-field jetpack-field-multiple"
-					label={
-						<JetpackFieldLabel
-							required={ required }
-							label={ label }
-							setAttributes={ setAttributes }
+					style={ {
+						marginBottom: spacing + 'px',
+					} }
+				>
+					{ options.map( ( option, index ) => (
+						<JetpackOption
+							type={ type }
+							key={ index }
+							option={ option }
+							index={ index }
+							onChangeOption={ onChangeOption }
+							onAddOption={ addNewOption }
+							isInFocus={ index === inFocus && isSelected }
 							isSelected={ isSelected }
-							resetFocus={ () => this.setState( { inFocus: null } ) }
 						/>
-					}
-				>
-					<ol
-						className="jetpack-field-multiple__list"
-						id={ `jetpack-field-multiple-${ instanceId }` }
+					) ) }
+				</ol>
+				{ isSelected && (
+					<IconButton
+						className="jetpack-field-multiple__add-option"
+						icon="insert"
```

Comment:
This seems to be different from the new block inserter icons:
<img width="697" alt="Screenshot 2020-04-24 at 15 07 18" src="https://user-images.githubusercontent.com/275961/80221517-4cedc780-863d-11ea-8f72-24e9403edccc.png">

Reply:
I think this is okay? I don't believe this should be the same icon, it's too confusing.